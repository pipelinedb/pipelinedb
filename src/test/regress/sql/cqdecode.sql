-- Decode all boolean literal types.

SET debug_sync_stream_insert = 'on';

CREATE CONTINUOUS VIEW test_bool AS SELECT b::boolean FROM stream;

ACTIVATE test_bool;

INSERT INTO stream (b) VALUES (TRUE), (true), ('true'), ('t'), ('y'), ('on'), ('1'), (1);
INSERT INTO stream (b) VALUES (FALSE), (false), ('false'), ('f'), ('n'), ('off'), ('0'), (0);

DEACTIVATE;

SELECT b, COUNT(*) FROM test_bool GROUP BY b;

DROP CONTINUOUS VIEW test_bool;
