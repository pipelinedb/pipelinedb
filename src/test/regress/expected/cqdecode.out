-- Decode all boolean literal types.
SET debug_sync_stream_insert = 'on';
CREATE CONTINUOUS VIEW test_bool AS SELECT b::boolean FROM cqdecode_stream;
ACTIVATE test_bool;
INSERT INTO cqdecode_stream (b) VALUES (TRUE), (true), ('true'), ('t'), ('y'), ('on'), ('1'::boolean), (1::boolean);
INSERT INTO cqdecode_stream (b) VALUES (FALSE), (false), ('false'), ('f'), ('n'), ('off'), ('0'::boolean), (0::boolean);
DEACTIVATE test_bool;
SELECT b, COUNT(*) FROM test_bool GROUP BY b;
 b | count 
---+-------
 f |     8
 t |     8
(2 rows)

DROP CONTINUOUS VIEW test_bool;
